DigitalLearningSolutions.Data.Tests/Services/PostLearningAssessmentTests.cs

```diff
@@ -0,0 +1,178 @@
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_return_null_if_customisation_id_is_invalid()
+        {
+            // Given
+            const int customisationId = 0;
+            const int candidateId = 84238;
+            const int sectionId = 103;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_return_null_if_section_id_is_invalid()
+        {
+            // Given
+            const int customisationId = 6062;
+            const int candidateId = 84238;
+            const int sectionId = 0;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_return_null_if_archived_date_is_null()
+        {
+            // Given
+            const int customisationId = 14212;
+            const int candidateId = 23031;
+            const int sectionId = 261;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_return_null_if_pl_assess_path_is_null()
+        {
+            // Given
+            const int customisationId = 809;
+            const int candidateId = 86254;
+            const int sectionId = 16;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_return_null_if_customisation_is_not_assessed()
+        {
+            // Given
+            const int customisationId = 5988;
+            const int candidateId = 78541;
+            const int sectionId = 16;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [TestCase(106456, 7561, 112, 75513)]
+        [TestCase(172807, 9917, 112, 133084)]
+        [TestCase(160541, 11698, 198, 133094)]
+        [TestCase(208119, 4918, 136, 167637)]
+         public void Get_post_learning_assessment_should_have_same_scores_as_stored_procedure(
+            int candidateId,
+            int customisationId,
+            int sectionId,
+            int progressId
+        )
+        {
+            using (new TransactionScope())
```

Comment:
Does this need to be in transaction scope? For other tests like this I've added it since I made it enrol the user if they weren't enrolled. But it doesn't look like we're doing that here? 🙂